playbooks/common/openshift-cluster/restart.yml

```diff
@@ -0,0 +1,161 @@
+---
+- include: evaluate_groups.yml
+
+- name: Validate configuration for rolling restart
+  hosts: oo_masters_to_config
+  roles:
+  - openshift_facts
+  tasks:
+  - fail:
+      msg: "openshift_rolling_restart_mode must be set to either 'services' or 'system'"
+    when: openshift_rolling_restart_mode is defined and openshift_rolling_restart_mode not in ["services", "system"]
+  - openshift_facts:
+      role: "{{ item.role }}"
+      local_facts: "{{ item.local_facts }}"
+    with_items:
+      - role: common
+        local_facts:
+          rolling_restart_mode: "{{ openshift_rolling_restart_mode | default('services') }}"
+      - role: master
+        local_facts:
+          cluster_method: "{{ openshift_master_cluster_method | default(None) }}"
+
+# Creating a temp file on localhost, we then check each system that will
+# be rebooted to see if that file exists, if so we know we're running
+# ansible on a machine that needs a reboot, and we need to error out.
```

Comment:
Do we want to error out here, or pause with a message stating that the localhost will be skipped (or performed last and not verified)?

That said, we should probably break out restarting the different host types into separate playbooks (and put them under the associated playbooks/common directories for those components) and have this playbook include those playbooks to perform the restart tasks...  maybe even flags to skip restarting certain host types or to only restart a single host type?